paddle/fluid/memory/allocation/buffered_allocator.h

```diff
@@ -31,7 +31,7 @@ namespace allocation {
 // underlying_allocator_
 class BufferedAllocator : public Allocator {
  public:
-  explicit BufferedAllocator(std::unique_ptr<Allocator> &&allocator);
+  explicit BufferedAllocator(std::shared_ptr<Allocator> allocator);
```

Comment:
Because some allocator accepts `std::shared_ptr<Allocator>` when constructing. We can cast `std::unique_ptr<Allocator>` to `std::shared_ptr<Allocator>` easily, but it is impossible to do the reverse casting.